deepchecks/checks/distribution/whole_dataset_drift.py

```diff
@@ -0,0 +1,234 @@
+        }
+
+        headnote = f"""<span>
+                    The shown features are the features that are most important for the domain classifier - the model
+                    trained to distinguish between the train and test datasets.<br> The percents of explained dataset 
+                    difference are the calculated feature importance values for the feature.
+                </span>"""
+
+        top_fi = fi_ser.head(self.n_top_features)
+        top_fi = top_fi.loc[top_fi > self.min_feature_importance]
+
+        displays = [headnote] +\
+                   [partial(self.display_dist, train_sample_df[feature], test_sample_df[feature], fi_ser)
+                    for feature in top_fi.index]
+
+        return CheckResult(value=values_dict, display=displays, header='Whole Dataset Drift')
+
+    def display_dist(self, train_column: pd.Series, test_column: pd.Series, fi_ser: pd.Series):
+        colors = ['darkblue', '#69b3a2']
+
+        plt.figure(figsize=(8, 3))
+        axs = plt.gca()
+
+        column_name = train_column.name
+
+        if column_name in self._cat_features:
+            expected_percents, actual_percents, categories_list = \
+                preprocess_for_psi(dist1=train_column.dropna().values.reshape(-1),
+                                   dist2=test_column.dropna().values.reshape(-1),
+                                   max_num_categories=self.max_num_categories)
+
+            cat_df = pd.DataFrame({'Train dataset': expected_percents, 'Test dataset': actual_percents},
+                                  index=categories_list)
+
+            cat_df.plot.bar(ax=axs, color=colors)
+            axs.set_ylabel('Percentage')
+            axs.legend()
+            plt.xticks(rotation=30)
+
+        else:
+            x_range = (min(train_column.min(), test_column.min()), max(train_column.max(), test_column.max()))
+            xs = np.linspace(x_range[0], x_range[1], 40)
+            pdf1 = plot_density(train_column, xs, colors[0])
+            pdf2 = plot_density(test_column, xs, colors[1])
+            plt.gca().set_ylim(bottom=0, top=max(max(pdf1), max(pdf2)) * 1.1)
+            axs.set_xlabel(column_name)
+            axs.set_ylabel('Probability Density')
+            axs.legend(['Train dataset', 'Test Dataset'])
+
+        plt.title(f'Feature: {column_name} - Explains {format_percent(fi_ser.loc[column_name])} of dataset difference')
+
+    def _generate_model(self, numerical_columns: List[Hashable], categorical_columns: List[Hashable]) -> Pipeline:
+
+        class PandasSimpleImputer(SimpleImputer):
+            """A wrapper around `SimpleImputer` to return data frames with columns.
+            """
+
+            def fit(self, x, y=None):
+                self.columns = x.columns
+                return super().fit(x, y)
+
+            def transform(self, x):
+                return pd.DataFrame(super().transform(x), columns=self.columns)
```

Comment:
Why not put in dist_utils? Weird to me that you define a useful class inside a private function